### ISSUE TITLE
MongoDB source: Object fields should be serialized as JSON

### ISSUE THREAD

**Issue by akoshterek** [airbytehq/airbyte#3462]
## Expected Behavior
I have a collection with Object type field.
It should be serialized as a proper JSON string like
`{"_id" : "609a228a84ce3e141d97e1df", "field": "value", "another_field": "another_value"}`

## Current Behavior
It appears in destination like
`{"_id"=>BSON::ObjectId('609a228a84ce3e141d97e1df'), "field"=>"value", "another_field"=>"another_value"}`
Also timestamp strings appear without quotes like
`"last_updated"=>2021-05-10 09:53:25 UTC`

The problematic collection has 7 records at the moment so schema discovery should not be an issue
![Screenshot 2021-05-18 at 11 51 22](https://user-images.githubusercontent.com/13558863/118637433-16e62d00-b7d6-11eb-834f-962d5c38ee4f.png)
![](https://user-images.githubusercontent.com/13558863/118637446-1a79b400-b7d6-11eb-82a3-e7bdc8b3df58.png)

It looks like an issue in mongodb_types_converter.rb

```
    when MongodbStream::AIRBYTE_TYPES[:object]
          value.is_a?(Hash) ? value : { 'value' => value.to_s }
```

possibly it should be 

```
    when MongodbStream::AIRBYTE_TYPES[:object]
          value.is_a?(Hash) ? value : { 'value' => value.to_json }
```

according to https://www.rubydoc.info/github/mongodb/bson-ruby/BSON/JSON

## Steps to Reproduce
1. Create a collection with an Object field with some nested fields
1. Run sync and check the data in destination

## Severity of the bug for you
High, as extracting data with SQL from not a JSON string is more complicated

## Airbyte Version
0.22.2-alpha

## Connector Version (if applicable)
MongoDB 0.3.1 -> BQ 0.3.2

## Additional context
Slack thread: https://airbytehq.slack.com/archives/C01MFR03D5W/p1621330354005000

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368131318955) by [Unito](https://www.unito.io)

**Comment by marcosmarxm** [comment]
I added the issue to the connector sprint, any chances to try to implement the solution @akoshterek? No pressure on that but I can give support setting things up.

**Comment by akoshterek** [comment]
I may try it next week

**Comment by akoshterek** [comment]
Unfortunately my assumption was wrong. I added logging into every branch and found that strings like {"_id"=>BSON::ObjectId('609a228a84ce3e141d97e1df'), "field"=>"value", "another_field"=>"another_value"} indeed arrive into MongodbTypesConverter as strings and not as objects.

location is indeed a very complex field, containing 2 levels of nested collections and it may confuse schema discovery mechanism I guess

Did a schema export in IntelliJ client
cdrs: table
+ columns
_id: ObjectId
_class: String
auth_id: String
auth_method: String
billed_partner_id: String
charging_periods: list
currency: String
external_id: String
first_received_at: ISODate
last_updated: ISODate
location: Object
owner_id: String
partner_id: String
received_at: ISODate
schema_version: Int32
start_date_time: ISODate
stop_date_time: ISODate
tariffs: list
total_cost: Decimal128
version: Int64
charging_periods.dimensions: list
location._id: ObjectId
total_energy: Decimal128
charging_periods.dimensions.type: String
location.address: String
total_time: Decimal128
charging_periods.dimensions.volume: Decimal128
location.charging_when_closed: Boolean
charging_periods.start_date_time: ISODate
location.city: String
location.coordinates: Object
location.coordinates.coordinates: Array
location.coordinates.type: String
location.country: String
location.directions: list
location.evses: list
location.evses.capabilities: Array
location.evses.connectors: list
location.evses.connectors.amperage: Int32
location.evses.connectors.format: String
location.evses.connectors.id: String
location.evses.connectors.last_updated: ISODate
location.evses.connectors.power_type: String
location.evses.connectors.standard: String
location.evses.connectors.tariff_id: String
location.evses.connectors.voltage: Int32
location.evses.coordinates: Object
location.evses.coordinates.coordinates: Array
location.evses.coordinates.type: String
location.evses.directions: list
location.evses.evse_id: String
location.evses.images: list
location.evses.last_updated: ISODate
location.evses.status: String
location.evses.uid: String
location.external_id: String
location.images: list
location.last_updated: ISODate
location.operator: Object
location.operator.name: String
location.owner_id: String
location.partner_id: String
location.postal_code: String
location.schema_version: Int32
location.type: String

**Comment by akoshterek** [comment]
I will ask guys to try

**Comment by akoshterek** [comment]
@sherifnada 

It's better now, not perfect though and perhaps it would be very difficult to solve it completely in a generic way.
I've got the comment: 

> Arrays are actually included now, albeit as repeated strings. Fields that have null values are not included in those strings.
> It doesn't create tables for nested objects.

However we've workarounded it already with using custom connector, which accepts array of JSON schemas. With it documents are fully parsed and basic normalisation handles them perfectly. Perhaps it worth closing.

![image](https://user-images.githubusercontent.com/13558863/139702028-925371fc-4bc8-48ba-8ec5-777856095e49.png)
